### DISCUSSION TITLE
Instant prompt

### DISCUSSION THREAD

**Discussion by fedoranvar** [nushell/nushell#4750] [Q&A]
Good day to you!

nushell without config start almost instantly, but if i'm adding some aliases or define function it increases startup time to ~200-300 ms (became noticed to the eye).

ZSH community has a hack for showing instantly prompt, is there any possibility for nushell, to instantly show prompt and keep loading everything else in background?

Thank you!

**Comment by sophiajt** [comment]
@fedoranvar - We've been doing work on speeding up the Nushell startup. You can see some of this work in the latest version in main, which will be part of the upcoming 0.60 release.

**Comment by sophiajt** [comment]
@fedoranvar - We've been doing work on speeding up the Nushell startup. You can see some of this work in the latest version in main, which will be part of the upcoming 0.60 release.
